File: Programmers/JS/완전탐색_카펫.js
function solution(brown, yellow) {
  var answer = [];
  var total = brown + yellow;
  function getNumber(num, total) {
    var otherSide = total / num;
    var checkBrown = 2 * (num + otherSide) - 4;
    var checkYellow = total - checkBrown;
    return [checkBrown, checkYellow];
  }
  for (var i = 2; i < total; i++) {
    if (total % i === 0) {
      var numSet = getNumber(i, total);
      console.log(numSet);
      if (numSet[0] === brown && numSet[1] === yellow) {
        answer[1] = total / i;
        answer[0] = i;
      }
    }
  }
  return answer;
}

File: Programmers/Python/완전탐색_카펫.py
def getBrown(x, y):
    res = 2 * (x+y) - 4
    return res

def solution(brown, yellow):
    answer = []
    total = brown + yellow
    for i in range(1, total):
        if total%i == 0:
            y = i
            x = total//i
            br = getBrown(x, y)
            if br == brown:
                answer.append(x)
                answer.append(y)
                break
    return answer